accounts/views.py

```diff
@@ -143,4 +94,33 @@ def test_token(request):
         json: validation msg
     """
     print(request.user)
-    return Response("passed for {}".format(request.user.username))
\ No newline at end of file
+    return Response("passed for {}".format(request.user.username))
+
+@api_view(['GET'])
+def get_user_by_username(request, username):
+    try:
+        user = User.objects.get(username=username)
+        serializer = UserPublicSerializer(user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    except User.DoesNotExist:
+        return Response({"detail": "User not found"}, status=status.HTTP_404_NOT_FOUND)
+    
+@api_view(['GET'])
+@authentication_classes([SessionAuthentication, TokenAuthentication])
+@permission_classes([IsAuthenticated])
+def get_user_personal_profile(request, username):
```

Comment:
The name of this function isn't as clear as it could be.
What about `get_logged_user`, `get_authenticated_user` or something like that?

Reply:
I've changed other function to get_user_public_profile() .....cos I prefer some kind of consistency now there are 2 views for 2 endpoints:
- get_user_public_profile ----> for public view
- get_user_private_profile ----> for auth user.

Reply:
Better talk about that 😂

Reply:
I also think the name of the function `get_logged_user` is better in this case. I agree with @milojezek.